app/assets/javascripts/initializers/initializeCommentPreview.js

```diff
@@ -1,50 +1,55 @@
-function initializeCommentPreview() {
-  if (document.getElementById('preview-button')) {
-    document.getElementById('preview-button').onclick = handleCommentPreview;
-  }
-}
-
-function handleCommentPreview(e) {
-  e.preventDefault();
-  var form = e.target.parentElement.parentElement;
-  var textArea = form.querySelector('textarea');
-  if (textArea.value !== '') {
-    var previewDiv = form.querySelector('.comment-preview-div');
-    var previewButton = form.querySelector('.comment-action-preview');
-    if (previewButton.innerHTML.indexOf('PREVIEW') > -1) {
-      textArea.classList.toggle('preview-loading');
-      getAndShowPreview(previewDiv, textArea);
-      previewButton.innerHTML = 'MARKDOWN';
-    } else {
-      previewDiv.classList.toggle('preview-toggle');
-      textArea.classList.toggle('preview-toggle');
-      previewButton.innerHTML = 'PREVIEW';
-    }
+function getAndShowPreview(markdownPreviewPane, markdownEditor) {
+  function successCb(body) {
+    markdownPreviewPane.classList.toggle('preview-toggle');
+    markdownEditor.classList.toggle('preview-loading');
+    markdownEditor.classList.toggle('preview-toggle');
+    markdownPreviewPane.innerHTML = body.processed_html; // eslint-disable-line no-param-reassign
   }
-}
 
-function getAndShowPreview(previewDiv, textArea) {
-  var commentMarkdown = textArea.value;
-  var payload = JSON.stringify({
+  const payload = JSON.stringify({
     comment: {
-      body_markdown: commentMarkdown,
+      body_markdown: markdownEditor.value,
     },
   });
   getCsrfToken()
     .then(sendFetch('comment-preview', payload))
-    .then(function(response) {
+    .then(response => {
```

Comment:
This (arrow function) totally breaks IE11 IIRC because this code is not transpiled. I'm okay with the general idea because it's not "supported" but I don't know if not supported explicitly means breaking the site with IE11 or just gracefully degrading on it. 

The same goes for destructuring.

It's probably part of a larger conversation about how to refactor this code.

I'm tagging @benhalpern and @aspittel because they likely have the correct answer here.

In the meantime, thanks a lot of taking care of the code :-)